projects/rocprofiler-systems/source/lib/common/environment.hpp

```diff
@@ -265,5 +265,81 @@ discover_llvm_libdir_for_ompt(bool verbose = false)
     return {};
 }
 
+enum update_mode : uint8_t
```

Comment:
```suggestion
enum class update_mode : uint8_t
```

And if you want to combine them

```
inline update_mode operator|(update_mode lhs, update_mode rhs) {
    using T = std::underlying_type_t<update_mode>;
    return static_cast<update_mode>(
        static_cast<T>(lhs) | static_cast<T>(rhs)
    );
}

inline update_mode operator&(update_mode lhs, update_mode rhs) {
    using T = std::underlying_type_t<update_mode>;
    return static_cast<update_mode>(
        static_cast<T>(lhs) & static_cast<T>(rhs)
    );
}
```

Reply:
Applied `enum class` as suggested.

Did **not** add `operator|` / `operator&` for combining modes.

Per Alex comment, modes should be **mutually exclusive** (exactly one of REPLACE, PREPEND, APPEND, or WEAK). Adding bitwise operators would allow ambiguous combinations like `WEAK | APPEND`, which we want to avoid.